crates/circuit/src/circuit_data.rs

```diff
@@ -1444,6 +1446,52 @@ impl CircuitData {
         Ok(())
     }
 
+    /// Append a standard gate to this CircuitData
+    pub fn push_standard_instruction(
+        &mut self,
+        operation: StandardInstruction,
+        params: &[Param],
+        qargs: &[Qubit],
+        cargs: &[Clbit],
+    ) -> PyResult<()> {
+        let params = (!params.is_empty()).then(|| Box::new(params.iter().cloned().collect()));
+        let qubits = self.qargs_interner.insert(qargs);
+        let clbits = self.cargs_interner.insert(cargs);
+        self.data.push(PackedInstruction {
+            op: operation.into(),
+            qubits,
+            clbits,
+            params,
+            label: None,
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceLock::new(),
+        });
+        Ok(())
+    }
+
+    /// Append a py gate to this CircuitData
+    pub fn push_py_gate(
+        &mut self,
+        operation: PyGate,
+        params: &[Param],
+        qargs: &[Qubit],
+        cargs: &[Clbit],
+    ) -> PyResult<()> {
+        let params = (!params.is_empty()).then(|| Box::new(params.iter().cloned().collect()));
+        let qubits = self.qargs_interner.insert(qargs);
+        let clbits = self.cargs_interner.insert(cargs);
+        self.data.push(PackedInstruction {
+            op: operation.into(),
+            qubits,
+            clbits,
+            params,
+            label: None,
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceLock::new(),
+        });
+        Ok(())
+    }
```

Comment:
I think `CircuitData::push_packed_operation` does all of these things here too. Since you have either a `StandardGate` or a `PyGate`, all you'd need to do is call `operation.into()` as you send either as an argument.

Reply:
Thanks! At the time that I wrote the above, I did not realize that this is possible. I have removed these functions in [c6376f4](https://github.com/Qiskit/qiskit/pull/13929/commits/c6376f408739b1f23d16ccace75a1875d9d3b132).